### PR TITLE
ADD speech_recognition example

### PR DESCRIPTION
**Pull request by Soonhwan-Kwon**
This example based on DeepSpeech2 of Baidu helps you to build Speech-To-Text (STT) models at scale using

- CNNs, fully connected networks, (Bi-) RNNs, (Bi-) LSTMs, and (Bi-) GRUs for network layers,
- batch-normalization for training efficiency,
- and a Baidu's WarpCTC for loss calculations.

In order to make your own STT models, besides, all you need is to just edit a configuration file not actual codes.

### CODE REVIEW

File: example/speech_recognition/LICENSE [hunk 1]
```diff
@@ -0,0 +1,204 @@
+Copyright 2015-2016 Baidu USA LLC.  All rights reserved.
+Modifications copyright (C) 2017 Soonhwan Kwon, Kyusam Oh, Min Soo Kim, Jihyeon Seo, Junyoung Park, Seungkyung Kim, and Wanggeun Park
+
+                                 Apache License
+                           Version 2.0, January 2004
```
Comment (piiswrong) [inline example/speech_recognition/LICENSE]: apache 2.0 license is implicit.

**Comment by piiswrong** [comment]
Thanks for the contribution!
Do you have acc numbers on standard datasets? Does this reproduce baidu's results?

**Comment by piiswrong** [comment]
FYI, we now have a mx.rnn package and rnn.FusedRNNCell uses cudnn rnn and might improve speed a lot

**Comment by sbodenstein** [comment]
One other thing: this should be merged soon (https://github.com/dmlc/mxnet/pull/5834), which will simplify your code a bit as: 

- the new CTC loss operator returns the actual CTC loss. So there is no need to write your own much slower Python implementation in `stt_metric.py` 
- don't need instructions to the user on building WarpCTC 
- your implementation will work on Windows (your current implementation relies on WarpCTC, which doesn't have a Windows build script)

**Comment by yzhang87** [comment]
Nice work! Do you have numbers on librispeech?

**Comment by Soonhwan-Kwon** [comment]
@piiswrong @sbodenstein Our team is very serious on this example and we are happy to keep up with new updates.

**Comment by Soonhwan-Kwon** [comment]
@piiswrong And we used native network for the consideration of batchnorm. How can I implement batchnorm in mx.rnn and rnn.FusedCell?

**Comment by piiswrong** [comment]
if you are only applying batchnorm on output then you can used fusernncell by stacking them and insert bn in between.

if you are doing batchnorm on states between time steps then you can no longer use fused rnn cell

**Comment by Soonhwan-Kwon** [comment]
@piiswrong @yzhang87 We are working on the accuracy, and will provide the result.

File: example/speech_recognition/LICENSE (hunk 1 above)
Comment (Soonhwan-Kwon) [inline example/speech_recognition/LICENSE]: @piiswrong updated LICENSE file to state where we modified

Comment (Soonhwan-Kwon) [inline example/speech_recognition/LICENSE]: @piiswrong would you tell me more details of implicity? I'm afraid that I'm confused about what you meant.
